scripts/exregional_run_prdgen.sh

```diff
@@ -0,0 +1,364 @@
+# from CDATE.
+#
+#-----------------------------------------------------------------------
+#
+yyyymmdd=${PDY}
+hh=${cyc}
+#-----------------------------------------------------------------------
+#
+# A separate ${post_fhr} forecast hour variable is required for the post
+# files, since they may or may not be three digits long, depending on the
+# length of the forecast.
+#
+#-----------------------------------------------------------------------
+#
+
+len_fhr=${#fhr}
+if [ ${len_fhr} -eq 9 ]; then
+  post_min=${fhr:4:2}
+  if [ ${post_min} -lt 15 ]; then
+    post_min=00
+  fi
+else
+  post_min=00
+fi
+
+subh_fhr=${fhr}
+if [ ${len_fhr} -eq 2 ]; then
+  post_fhr=${fhr}00
+elif [ ${len_fhr} -eq 3 ]; then
+  if [ "${fhr:0:1}" = "0" ]; then
+    post_fhr="${fhr:1}00"
+  else
+    post_fhr=${fhr}00
+  fi
+elif [ ${len_fhr} -eq 9 ]; then
+  if [ "${fhr:0:1}" = "0" ]; then
+    if [ ${post_min} -eq 00 ]; then
+      post_fhr="${fhr:1:2}00"
+      subh_fhr="${fhr:0:3}"
+    else
+      post_fhr="${fhr:1:2}${fhr:4:2}"
+    fi
+  else
+    if [ ${post_min} -eq 00 ]; then
+      post_fhr="${fhr:0:3}00"
+      subh_fhr="${fhr:0:3}"
+    else
+      post_fhr="${fhr:0:3}${fhr:4:2}"
+    fi
+  fi
+else
+  print_err_msg_exit "\
+The \${fhr} variable contains too few or too many characters:
+  fhr = \"$fhr\""
+fi
+
+# replace fhr with subh_fhr
+echo "fhr=${fhr} and subh_fhr=${subh_fhr}"
+fhr=${subh_fhr}
+
+#
+# background files
+#
+prslev=${COMOUT}/${NET}.${cycle}${dot_ensmem}.prslev.f${fhr}.${POST_OUTPUT_DOMAIN_NAME}.grib2
+natlev=${COMOUT}/${NET}.${cycle}${dot_ensmem}.natlev.f${fhr}.${POST_OUTPUT_DOMAIN_NAME}.grib2
+testbed=${COMOUT}/${NET}.${cycle}${dot_ensmem}.testbed.f${fhr}.${POST_OUTPUT_DOMAIN_NAME}.grib2
+ififip=${COMOUT}/${NET}.${cycle}${dot_ensmem}.ififip.f${fhr}.${POST_OUTPUT_DOMAIN_NAME}.grib2
+
+# extract the output fields for the testbed files
+touch ${testbed}
+if [[ ! -z ${TESTBED_FIELDS_FN} ]]; then
+  if [[ -f "${PARMdir}/upp/${TESTBED_FIELDS_FN}" ]]; then
+    wgrib2 ${prslev} | grep -F -f "${PARMdir}/upp/${TESTBED_FIELDS_FN}" | wgrib2 -i -grib ${testbed} ${prslev}
+  else
+    echo "${PARMdir}/upp/${TESTBED_FIELDS_FN} not found"
+  fi
+fi
+if [[ ! -z ${TESTBED_FIELDS_FN2} ]]; then
+  if [[ -f "${PARMdir}/upp/${TESTBED_FIELDS_FN2}" ]]; then
+    wgrib2 ${natlev} | grep -F -f "${PARMdir}/upp/${TESTBED_FIELDS_FN2}" | wgrib2 -i -append -grib ${testbed} ${natlev}
+  else
+    echo "${PARMdir}/upp/${TESTBED_FIELDS_FN2} not found"
+  fi
+fi
+
+gridname=""
+if [ ${PREDEF_GRID_NAME} = "RRFS_CONUS_3km" ]; then
+  gridname="conus_3km."
+elif  [ ${PREDEF_GRID_NAME} = "RRFS_NA_3km" ]; then
+  gridname=""
+fi
+basetime=$( date +%y%j%H%M -d "${yyyymmdd} ${hh}" )
+
+# Create index (idx) files
+net4=$(echo ${NET:0:4} | tr '[:upper:]' '[:lower:]')
+for leveltype in prslev natlev ififip testbed
+do
+  if [ -f ${COMOUT}/${NET}.${cycle}${dot_ensmem}.${leveltype}.f${fhr}.${POST_OUTPUT_DOMAIN_NAME}.grib2 ]; then
```

Comment:
I have not tested the naming of files with "ensemble" members properly -- because RRFS CONUS 3km did not have it IIRC. So the `${dot_ensmem} additions are not tested although I have tested `f${fhr}` which was represented as `{fhr}` works